File: backend/src/models/Volunteer.js
import mongoose from "mongoose";

const volunteerSchema = new mongoose.Schema({
  name: { type: String, required: true },
  email: { type: String, required: true },
  contact: { type: String, required: true },
  address: { type: String, required: true },
  district: { type: String, required: true },
  skills: { type: String, required: true },
  lat: { type: Number },
  lng: { type: Number },
  createdAt: { type: Date, default: Date.now },
});

const Volunteer = mongoose.model("Volunteer", volunteerSchema);

export default Volunteer;

File: backend/models/volunteer.py
from mongoengine import Document, StringField, DateTimeField, signals
from datetime import datetime
import pytz

class Volunteer(Document):
    name = StringField(required=True)
    contact = StringField()
    address = StringField()
    district = StringField()
    timeanddate = DateTimeField()  # Automatically updated on save

    @classmethod
    def pre_save(cls, sender, document, **kwargs):
        """Update the timeanddate field before saving with Indian timezone."""
        india_tz = pytz.timezone('Asia/Kolkata')
        document.timeanddate = datetime.now(india_tz)

signals.pre_save.connect(Volunteer.pre_save, sender=Volunteer)